packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,739 @@
+			};
+		case TestPrimitives.Boolean:
+			return { value: random.pick(booleans), type: dataSchema.name };
+		case TestPrimitives.Map:
+			map = {
+				mapField2: {
+					mapField3: [{ type: dataSchema.name, value: insertValue.toString() }],
+				},
+			};
+			return {
+				value: map,
+				type: dataSchema.name,
+			};
+		default:
+			unreachableCase(dataType);
+	}
+}
+
+function getTestTreeAsJSObject(
+	numberOfNodes: number,
+	shape: TreeShape,
+	dataType: TestPrimitives,
+): any {
+	const seed = 0;
+	const random = makeRandom(seed);
+	let tree;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree = [getJSTestTreeDeep(numberOfNodes, dataType, random)];
+			break;
+		case TreeShape.Wide:
+			tree = getJSTestTreeWide(numberOfNodes, dataType, random);
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	const testTreeJS = JSON.parse(JSON.stringify(tree));
+	return testTreeJS;
+}
+
+function getJSTestTreeWide(numberOfNodes: number, dataType: TestPrimitives, random: IRandom): any {
+	const nodes = [];
+	for (let i = 0; i < numberOfNodes - 1; i++) {
+		const node = generateTreeData(dataType, random);
+		nodes.push(node);
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: nodes,
+		},
+		value: generateTreeData(dataType, random).value,
+	};
+	return tree;
+}
+
+function getJSTestTreeDeep(numberOfNodes: number, dataType: TestPrimitives, random: IRandom): any {
+	const node = generateTreeData(dataType, random);
+	if (numberOfNodes === 1) {
+		return {
+			type: dataSchema.name,
+			value: node.value,
+		};
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: [getJSTestTreeDeep(numberOfNodes - 1, dataType, random)],
+		},
+		value: node.value,
+	};
+	return tree;
+}
+
+function readTreeAsJSObject(tree: any) {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) readTreeAsJSObject(tree[key]);
+		if (key === "value") {
+			assert(tree[key] !== undefined);
+		}
+	}
+}
+
+function manipulateTreeAsJSObject(tree: any, dataType: TestPrimitives, random: IRandom): any {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null)
+			manipulateTreeAsJSObject(tree[key], dataType, random);
+		if (key === "value") {
+			tree[key] = generateTreeData(dataType, random).value;
+		}
+	}
+}
+
+function readCursorTree(forest: any, numberOfNodes: number, shape: TreeShape) {
+	const readCursor = forest.allocateCursor();
+	moveToDetachedField(forest, readCursor);
+	assert(readCursor.firstNode());
+	switch (shape) {
+		case TreeShape.Deep:
+			for (let i = 0; i < numberOfNodes - 1; i++) {
+				readCursor.firstField();
+				assert(readCursor.firstNode());
+			}
+			break;
+		case TreeShape.Wide:
+			readCursor.firstField();
+			readCursor.firstNode();
+			for (let j = 0; j < numberOfNodes - 1; j++) {
+				readCursor.nextNode();
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	readCursor.free();
+}
+
+function manipulateTree(
```

Comment:
there are many ways a tree could be manipulated and its not clear which this does from its arguments. More documentation would be great.

Reply:
looks like this case makes numberOfNodes separate set value transactions using the editor DefaultEditBuilder.setValue API.

I'd be more interested in the performance of either:
1. making a large transaction
2. editing part of a large tree

Generally making O(size of tree) separate transactions isn't something we should do, and is also the only editing case we already have a test for (since its what bubble bench does)

Reply:
Added documentation, applies edit to only the final leaf node